inference-engine/src/inference_engine/include/openvino/runtime/executable_network.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides ExecutableNetwork class
+ *
+ * @file ie_executable_network.hpp
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "ie_parameter.hpp"
+#include "infer_request.hpp"
+
+namespace InferenceEngine {
+class IExecutableNetworkInternal;
+class RemoteContext;
+}  // namespace InferenceEngine
+namespace ov {
+class Function;
+namespace runtime {
+
+class Core;
+
+/**
+ * @brief This is an interface of an executable network
+ */
+class INFERENCE_ENGINE_API_CLASS(ExecutableNetwork) {
+    std::shared_ptr<SharedObject> _so;
+    std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> _impl;
+
+    /**
+     * @brief Constructs ExecutableNetwork from the initialized std::shared_ptr
+     * @param so Plugin to use. This is required to ensure that ExecutableNetwork can work properly even if plugin
+     * object is destroyed.
+     * @param impl Initialized shared pointer
+     */
+    ExecutableNetwork(const std::shared_ptr<SharedObject>& so,
+                      const std::shared_ptr<ie::IExecutableNetworkInternal>& impl);
+    friend class ov::runtime::Core;
+
+public:
+    /**
+     * @brief A default constructor.
+     */
+    ExecutableNetwork() = default;
+
+    /**
+     * @brief Creates an inference request object used to infer the network.
+     *
+     * The created request has allocated input and output blobs (that can be changed later).
+     *
+     * @return InferRequest object
+     */
+    InferRequest create_infer_request();
+
+    /**
+     * @brief Exports the current executable network.
+     *
+     * @see Core::ImportNetwork
+     *
+     * @param modelFileName Full path to the location of the exported file
+     */
+    void export_model(const std::string& modelFileName);
+
+    /**
+     * @brief Exports the current executable network.
+     *
+     * @see Core::ImportNetwork
+     *
+     * @param networkModel Network model output stream
+     */
+    void export_model(std::ostream& networkModel);
+
+    /**
+     * @brief Get executable graph information from a device
+     *
+     * @return Function containing Executable Graph Info
+     */
+    std::shared_ptr<const Function> get_function() const;
```

Comment:
I think we can introduce helpers:
`get_parameters()`, `get_results()`.
It should allow to simplify the code:
```
exec_network.get_parameters();
// instead of 
exec_network.get_function()->get_parameters();
```

Reply:
done